test/test.js

```diff
@@ -237,3 +237,38 @@ test('should check fastify dependency graph - decorateReply', t => {
     t.equal(err.message, "The decorator 'plugin2' required by 'test' is not present in Reply")
   })
 })
+
+test('should accept an option to encapsulate', t => {
+  t.plan(4)
+  const fastify = Fastify()
+
+  fastify.register(fp((fastify, opts, next) => {
+    fastify.decorate('accessible', true)
+    next()
+  }, {
+    name: 'accessible-plugin'
+  }))
+
+  fastify.register(fp((fastify, opts, next) => {
+    fastify.decorate('alsoAccessible', true)
+    next()
+  }, {
+    name: 'accessible-plugin2',
+    encapsulate: false
+  }))
+
+  fastify.register(fp((fastify, opts, next) => {
+    fastify.decorate('encapsulated', true)
+    next()
+  }, {
+    name: 'encapsulated-plugin',
+    encapsulate: true
```

Comment:
We need a test that verifies what the docs say: does it run all the validations?

Reply:
Would you say the 1 test is enough? I'd rather avoid repeating the whole test suite

Reply:
All the cases that have not been tested have an undetermined state, so the maintainers can't help whoever opens an issue.

In this case, the maintainer should add a test case for it and add it to the repository etc.. in an endless pain loop.

So, we can discuss about the minimal test suite we need, but yes: 1 test is not enough of a HUGE feature like this.

So I think the minimal test suite for this test suite is:

- check that the feature works as expected:
  - [x] are decorators encapsulated?
  - [ ] is the fastify version check working?
  - [ ] are mandatory decorators checks applied?
  - [ ] is the required plugins option working?
- verify how fastify reacts to this configuration (plugin name chaining)